code/game/mecha/equipment/passive/passive.dm

```diff
@@ -0,0 +1,59 @@
+/obj/item/mecha_parts/mecha_equipment/passive
+	is_activateable = FALSE
+
+/obj/item/mecha_parts/mecha_equipment/passive/rack
+	name = "\improper Exosuit-Mounted Rack"
+	desc = "A locking storage rack for the outside of an exosuit. (Can be attached to: Any exosuit)"
+	icon_state = "mecha_rack"
+	origin_tech = Tc_MATERIALS + "=3;" + Tc_MAGNETS + "=3;"
+	var/obj/item/weapon/storage/mechrack/rack
+
+/obj/item/mecha_parts/mecha_equipment/passive/rack/New()
+	..()
+	rack = new(src)
+	rack.mech_part = src
+
+/obj/item/mecha_parts/mecha_equipment/passive/rack/Destroy()
+	rack.empty_contents_to(chassis)
+	qdel(rack)
+	rack = null
+	..()
+
+/obj/item/mecha_parts/mecha_equipment/passive/rack/detach(atom/moveto=null)
+	rack.empty_contents_to(chassis)
+	..()
+
+/obj/item/weapon/storage/mechrack
+	name = "exosuit storage rack"
+	desc = "A large rack for an exosuit."
+	icon = 'icons/mecha/mecha_equipment.dmi'
+	icon_state = "mecha_rack"
+	fits_max_w_class = W_CLASS_LARGE
+	max_combined_w_class = 28
+	slot_flags = 0
+	rustle_sound = "rustle-metal"
+	var/obj/item/mecha_parts/mecha_equipment/mech_part
+
+/obj/item/weapon/storage/mechrack/Destroy()
+	mech_part = null
+	..()
+
+/obj/item/weapon/storage/mechrack/distance_interact(mob/user)
```

Comment:
Try testing it with telekinesis, just to be sure

Reply:
I actually factored this in already, but your comment made me realize that on line 44 it should be checking the range of `user,M`, not `src,M`, so good thought